src/frontend/src/components/authenticateBox.ts

```diff
@@ -0,0 +1,227 @@
+import { html, render, TemplateResult } from "lit-html";
+import { icLogo, caretDownIcon } from "./icons";
+import { footer } from "./footer";
+import { withLoader } from "./loader";
+import { addRemoteDevice } from "../flows/addDevice/welcomeView";
+import { mkAnchorInput } from "./anchorInput";
+import { getUserNumber, setUserNumber } from "../utils/userNumber";
+import { unreachable } from "../utils/utils";
+import { Connection } from "../utils/iiConnection";
+import { ref, createRef, Ref } from "lit-html/directives/ref.js";
+import {
+  apiResultToLoginFlowResult,
+  LoginFlowResult,
+  LoginFlowSuccess,
+  LoginFlowError,
+  LoginData,
+} from "../flows/login/flowResult";
+import { displayError } from "./displayError";
+import { useRecovery } from "../flows/recovery/useRecovery";
+import { registerIfAllowed } from "../utils/registerAllowedCheck";
+import { withRef } from "../utils/lit-html";
+
+/** Properties for the authentication box component which authenticates a user
+ * to II or to another dapp */
+type PageProps = { templates: AuthTemplates } & {
+  onContinue: (arg: bigint) => void;
```

Comment:
should `arg` be typed or be more explicit? just feel a bit generic

Reply:
ah it is! the type is `bigint`